core/domain/exp_services.py

```diff
@@ -237,6 +237,22 @@ def get_multiple_explorations_by_id(exp_ids, strict=True):
     return result
 
 
+def get_displayable_exploration_summaries_matching_ids(exploration_ids):
+        """Pass in list of exploration ids,
+        returns list with only existing, public explorations in list. 
+        """
+        displayable_exploration_summaries = []
+        exploration_summaries=get_exploration_summaries_matching_ids(
+            exploration_ids)
+
+        for exploration_summary in exploration_summaries:
+            if exploration_summary and exploration_summary.status == 'public':
```

Comment:
For the second part, use the check
 
   exploration_summary.status != rights_manager.ACTIVITY_STATUS_PRIVATE

instead. Two reasons: 'publicized' explorations are also public, and we should use named constants where possible.